client/clientfactory.go

```diff
@@ -60,7 +60,9 @@ type (
 		NewHistoryClientWithTimeout(timeout time.Duration) (historyservice.HistoryServiceClient, error)
 		NewMatchingClientWithTimeout(namespaceIDToName NamespaceIDToNameFunc, timeout time.Duration, longPollTimeout time.Duration) (matchingservice.MatchingServiceClient, error)
 		NewFrontendClientWithTimeout(rpcAddress string, timeout time.Duration, longPollTimeout time.Duration) workflowservice.WorkflowServiceClient
```

Comment:
rename this to `NewRemoteFrontedClient` or `NewExternalFrontClient` so it is more obvious?

Reply:
also I think we still need to fix `NewFrontendClientWithTimeout` not using the correct CERT when given external rpc endpoint (or address).

Reply:
Updated.